### ISSUE TITLE
Implement c/r error reporting

### ISSUE THREAD

**Issue by kolyshkin** [opencontainers/runc#3711]
Currently, when checkpoint or restore fails, runc diagnostics is inadequate, as the error itself is hiding in criu logs. This is why we have something like

```sh
		grep -B 5 Error "$workdir"/*.log ./image-dir/*.log || true
```

all across the https://github.com/opencontainers/runc/blob/main/tests/integration/checkpoint.bats

Perhaps we can implement something like this in runc itself, so it will show something useful in case of error.

**Comment by danishprakash** [comment]
@kolyshkin mind If I take a look at this?

**Comment by kolyshkin** [comment]
Fixed by #3816

**Comment by kolyshkin** [comment]
Somehow I forgot the last step, here it is: https://github.com/opencontainers/runc/pull/4251
